pkg/alexa/gen/skill.go

```diff
@@ -6,415 +6,283 @@ import (
-	s.countries = append(s.countries, c)
+func (s *SkillBuilder) AddCountry(country string) *SkillBuilder {
+	s.countries = append(s.countries, country)
+	return s
 }
 
-func (s *SkillBuilder) AddCountries(cs []alexa.Country) {
+func (s *SkillBuilder) AddCountries(cs []string) *SkillBuilder {
 	for _, c := range cs {
 		s.countries = append(s.countries, c)
 	}
+	return s
 }
 
 // Build builds an alexa.Skill object.
-// TODO: return errors!
 func (s *SkillBuilder) Build() (*alexa.Skill, error) {
+	if s.registry == nil || len(s.registry.GetLocales()) == 0 {
+		return nil, fmt.Errorf("No locales registered to build")
+	}
+	//
+	if s.locales == nil || len(s.locales) == 0 {
+		for n, _ := range s.registry.GetLocales() {
+			s.locales[n] = NewSkillLocaleBuilder(n).
+				WithLocaleRegistry(s.registry)
+
+		}
+	}
+
+	dl := s.registry.GetDefault()
+
 	skill := &alexa.Skill{
 		Manifest: alexa.Manifest{
-			Version: "1.0",
+			Version:    "1.0",
+			Publishing: alexa.Publishing{},
 		},
 	}
-
-	// default is always set if at least one locale was defined.
-	dl, _ := s.locales[s.defaultLocale]
-
 	skill.Manifest.Publishing.Category = s.category
 	// TODO: ensure unique occurance
-	skill.Manifest.Publishing.Countries = s.countries
-
-	if s.testingInstructions != "" {
-		skill.Manifest.Publishing.TestingInstructions = s.testingInstructions
+	if len(s.countries) > 0 {
+		skill.Manifest.Publishing.Countries = s.countries
 	} else {
-		skill.Manifest.Publishing.TestingInstructions = dl.Translations.Get(l10n.KeySkillTestingInstructions)
+		skill.Manifest.Publishing.Worldwide = true
 	}
+	skill.Manifest.Publishing.TestingInstructions = dl.Get(s.skillInstructions)
 
-	// Permissions are required.
+	// TODO: Permissions are required.
 	skill.Manifest.Permissions = &[]alexa.Permission{}
 
 	// PrivacyAndCompliance is required.
 	skill.Manifest.Privacy = &alexa.Privacy{}
-	if s.privacy.flags[FlagIsExportCompliant] {
+	if s.privacyFlags[FlagIsExportCompliant] {
 		skill.Manifest.Privacy.IsExportCompliant = true
 	}
-	if s.privacy.flags[FlagContainsAds] {
+	if s.privacyFlags[FlagContainsAds] {
 		skill.Manifest.Privacy.ContainsAds = true
 	}
-	if s.privacy.flags[FlagAllowsPurchases] {
+	if s.privacyFlags[FlagAllowsPurchases] {
 		skill.Manifest.Privacy.AllowsPurchases = true
 	}
-	if s.privacy.flags[FlagUsesPersonalInfo] {
+	if s.privacyFlags[FlagUsesPersonalInfo] {
 		skill.Manifest.Privacy.UsesPersonalInfo = true
 	}
-	if s.privacy.flags[FlagIsChildDirectred] {
+	if s.privacyFlags[FlagIsChildDirected] {
 		skill.Manifest.Privacy.IsChildDirected = true
 	}
 
 	// Add elements for every locale.
-	for _, l := range s.locales {
-		l.BuildLocale(skill)
-		l.BuildPrivacyLocale(skill)
+	skill.Manifest.Publishing.Locales = make(map[string]alexa.LocaleDef)
+	skill.Manifest.Privacy.Locales = make(map[string]alexa.PrivacyLocaleDef)
+	for n, _ := range s.registry.GetLocales() {
+		if l, err := s.locales[n].BuildPublishingLocale(); err != nil {
+			return nil, err
+		} else {
```

Comment:
dont use `else`. move the `l, err` assignment into its own line and handle the error.

Reply:
fixed